### PR TITLE
support passing fqdn to register cluster node in keeper

### PR DESCRIPTION
**Pull request by LiuYangkuan**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Support passing fqdn through setting `my_hostname` to register cluster node in keeper.
Add setting of `invisible` to support multi compute groups. A compute group as a cluster, 
is invisible to other compute groups.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
This is an automated comment for commit 6e80537ab61b95e5d6aa85a50532efc336ed4cdf with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/50042/6e80537ab61b95e5d6aa85a50532efc336ed4cdf/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Interpreters/ClusterDiscovery.cpp [hunk 1]
```diff
@@ -125,10 +125,12 @@ ClusterDiscovery::ClusterDiscovery(
             ClusterInfo(
                 /* name_= */ key,
                 /* zk_root_= */ config.getString(prefix + ".path"),
+                /* host_name= */ config.getString(prefix + ".my_hostname", getFQDNOrHostName()),
```
[the remark this answers is not in the document]
Comment (LiuYangkuan) [reply src/Interpreters/ClusterDiscovery.cpp:128]: In some k8s deployment environment, which not using core dns, and there are two network interface, eth0 and eth1, one is used by the k8s cluster control plane, the other eth1 is used by clickhouse business data plane. So, hostname can't resolve ip address of data plane. Therefore, we should explicitly specify the address of the host.

**Comment by LiuYangkuan** [comment]
> The PR looks good but could you expand on the usability of invisible and why it's necessary?

File: src/Interpreters/ClusterDiscovery.h [hunk 2]
```diff
@@ -78,16 +77,24 @@ class ClusterDiscovery
         /// Current node may not belong to cluster, to be just an observer.
         bool current_node_is_observer = false;
 
+        /// For internal management need.
+        /// Is it designed that when deploying multiple compute groups,
+        /// they are mutually invisible to each other.
+        bool current_cluster_is_invisible = false;
```
[the remark this answers is not in the document]
Comment (LiuYangkuan) [reply src/Interpreters/ClusterDiscovery.h:83]: A compute group contains lots of clickhouse servers, and there is mutual isolation of resources between computing groups. When we deploy multiple compute groups，the DDL should be executed all nodes of all compute groups.

**Comment by LiuYangkuan** [comment]
> The PR looks good but could you expand on the usability of invisible and why it's necessary?

Is it designed for multiple compute groups.
A compute group contains lots of clickhouse servers, and there is mutual isolation of resources between computing groups. When we deploy multiple compute groups，the DDL should be executed all nodes of all compute groups. But For a specific computing group, the node should register `default` and `all_nodes` cluster, with certain keeper path. But set invisible and this computing group can't use all_nodes cluster. Only the main computing group can use cluster `all_nodes` to dispatch ddl.

**Comment by LiuYangkuan** [comment]
I accidentally closed it, how can I reopen it? @antonio2368

**Comment by LiuYangkuan** [comment]
> I accidentally closed it, how can I reopen it? @antonio2368
I have merged branch master in this conversion, and force pushed after I accidentally closed it. Maybe can't reopen it.

**Comment by vdimir** [comment]
@LiuYangkuan 

For some reason button `Reopen` is unavailable (github says that branch was force-pushed). Could you just reopen new PR?

**Comment by LiuYangkuan** [comment]
> @LiuYangkuan
> 
> For some reason button `Reopen` is unavailable (github says that branch was force-pushed). Could you just reopen new PR?

OK, #50186 is new PR, thank you!

**Pull request closed without merging**